keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -88,15 +73,24 @@ class ArtifactListener(
     val artifact = event.artifact
 
     if (repository.artifactVersions(artifact).isEmpty()) {
-      when (artifact) {
-        is DebianArtifact -> storeLatestDebVersion(artifact)
-        is DockerArtifact -> storeLatestDockerVersion(artifact)
+      val artifactPublisher = artifactPublishers.supporting(artifact.type)
+      val latestArtifact = runBlocking {
+        log.debug("Retrieving latest version of registered artifact {}", artifact)
+        artifactPublisher.getLatestArtifact(artifact.deliveryConfig, artifact)
```

Comment:
something is not adding up for me -- how come `artifactPublisher` (which should only support `publish`), is getting the latest artifact?

Reply:
I mentioned this in other PRs. I went back-and-forth on the name for this class. My first name was `ArtifactSource`, then `ArtifactSupplier`, and finally (after a suggestion from @robzienert), `ArtifactPublisher`. I don't mind changing it again. Which one of those do you think is best? (Note, `ArtifactProvider` is already taken elsewhere in keel).

Reply:
What about `ArtifactPlugin`? (I don't care too much. Just throwing out an idea)